emcc.py

```diff
@@ -1472,6 +1475,11 @@ def check(input_file):
           # table, as is standard in wasm, and not asm.js split ones.
           shared.Settings.EMULATED_FUNCTION_POINTERS = 1
 
+    if shared.Settings.WASM2JS:
+      if not shared.Settings.WASM_BACKEND:
+        exit_with_error('wasm2js is only available in the upstream wasm backend path')
+      logger.warn('emcc: JS support in the upstream LLVM+wasm2js path is very experimental currently (best to use fastcomp for asm.js for now)')
```

Comment:
I think this is spelt "logger.warning" which is guess means this isn't tested.

Also, I don't think you need/want the `emcc:` prefix since the logger should give you that.

Reply:
Hmm, `logger.warn` works (and it is tested every time wasm2js is used). But the docs say

> Note There is an obsolete method warn which is functionally identical to warning. As warn is deprecated, please do not use it - use warning instead.

so I'll change it.